lib/form/paragraph3.flow

```diff
@@ -105,14 +110,29 @@ general2WrapElement(elem : GeneralWrapElement, style : [CharacterStyle]) -> Wrap
 	}
 }
 
+// Longest line width and all lines optimized
+getOptimizedLines(lines : [ParaLine], alignment : NumAlignment) -> Pair<double, [OptimizedLine]> {
+	lasti = length(lines) - 1;
+	// None() for Justify, last line alignment overridden.
+	optlines = mapi(lines, \i, line -> getOptimizedLine(line, i != lasti && alignment==Justify()));
+	Pair(fold(optlines, 0.0, \acc, optline -> max(acc, optline.metrics.width)), optlines)
+}
+
 renderParagraph(
 	p : [ParagraphElement],
 	s : [ParagraphStyle],
 	rtl : bool,
 	widthManager : WigiParagraphWidthManager
 ) -> Form {
 
-	alignment : ParaLineAlignment = extractParaLineAlignment(s);
+	alignment : NumAlignment = switch (extractParaLineAlignment(s)) {
+		StartAlign(): Some(0.0);
```

Comment:
I think it shows that the idea of double in align is not good.

Reply:
I simply moved this switch from RenderLine to upper level. NumAlignment type just introduced internally to pass the value used underneath. It looks different because width remainder available in RenderLine only, so left on place.